src/test/java/bio/terra/service/filedata/azure/tables/TableFileConnectedTest.java

```diff
@@ -85,4 +79,46 @@ public void testCreateDeleteEntry() {
     boolean isNotDeleted = tableFileDao.deleteFileMetadata(tableServiceClient, DATASET_ID, FILE_ID);
     assertFalse("File record was already deleted", isNotDeleted);
   }
+
+  @Test
+  public void testListEntry() {
+    List<FireStoreFile> fileList = IntStream.range(0, 5).boxed().map(this::makeFile).toList();
+    for (FireStoreFile fsFile : fileList) {
+      tableFileDao.createFileMetadata(tableServiceClient, DATASET_ID, fsFile);
+    }
+    String collectionId = FILES_TABLE.toTableName(UUID.fromString(DATASET_ID));
+    List<FireStoreFile> files =
+        tableFileDao.listFileMetadata(tableServiceClient, collectionId, 0, 10);
+    assertThat(files, equalTo(fileList));
+
+    List<FireStoreFile> filesOffset =
+        tableFileDao.listFileMetadata(tableServiceClient, collectionId, 1, 10);
+    assertThat(filesOffset.size(), equalTo(4));
```

Comment:
If you're asserting `equalTo()`, do you also need to assert the length? I think `equalTo()` will test that. If you do want to assert the length of a collection , I would use `hasSize()`:
```suggestion
    assertThat(filesOffset, hasSize(4));
```